CompanyEmployees.Infrastructure.Persistence/Repositories/AttendanceRepository.cs

```diff
@@ -0,0 +1,41 @@
+using CompanyEmployees.Core.Domain.Entities;
+using CompanyEmployees.Core.Domain.Repositories;
+using Microsoft.EntityFrameworkCore;
+
+namespace CompanyEmployees.Infrastructure.Persistence.Repositories;
+
+internal sealed class AttendanceRepository : RepositoryBase<Attendance>, IAttendanceRepository
+{
+    public AttendanceRepository(RepositoryContext repositoryContext)
+        : base(repositoryContext)
+    {
+    }
+
+    public async Task<IEnumerable<Attendance>> GetAllAttendancesAsync(bool trackChanges, CancellationToken ct = default) =>
+        await FindAll(trackChanges)
+            .OrderByDescending(a => a.CheckInTime)
+            .ToListAsync(ct);
+
+    public async Task<Attendance> GetAttendanceAsync(Guid attendanceId, bool trackChanges, CancellationToken ct = default) =>
+        await FindByCondition(a => a.Id.Equals(attendanceId), trackChanges)
+            .SingleOrDefaultAsync(ct);
+
+    public async Task<IEnumerable<Attendance>> GetAttendancesByEmployeeIdAsync(Guid employeeId, bool trackChanges, CancellationToken ct = default) =>
+        await FindByCondition(a => a.EmployeeId.Equals(employeeId), trackChanges)
+            .OrderByDescending(a => a.CheckInTime)
+            .ToListAsync(ct);
+
+    public async Task<Attendance?> GetTodayAttendanceByEmployeeIdAsync(Guid employeeId, bool trackChanges, CancellationToken ct = default)
+    {
+        var today = DateTime.Today;
+        return await FindByCondition(
+                a => a.EmployeeId.Equals(employeeId) && 
+                     a.CheckInTime.Date == today, 
+                trackChanges)
+            .FirstOrDefaultAsync(ct);
+    }
```

Comment:
To improve consistency, the return type of `GetAllAttendancesAsync` and `GetAttendancesByEmployeeIdAsync` could explicitly be marked as `IEnumerable<Attendance>` instead of relying on implicit lists. Although functionally correct, this clarifies intent.

**Suggested change**
```suggestion
public async Task<IEnumerable<Attendance>> GetAttendancesByEmployeeIdAsync(Guid employeeId, bool trackChanges, CancellationToken ct = default) =>
```